### PR TITLE
Navigation methods for VEC 1.2.0

### PR DESCRIPTION
**Pull request by DManstrator**
## Pull Request

- [x] I have checked for similar PRs.
- [x] I have read the [contributing guidelines](https://github.com/4Soft-de/vec-model/blob/develop/.github/CONTRIBUTING.md).

### Changes

- [x] Code
- [x] Documentation
- [ ] Other: 

### Description

Adding convenient navigation methods for the VEC 1.2.0. Also introduced a basic annotation `@RequiresBackReferences`
to mark methods. For such methods a manually created `VecContent` might result in exceptions since back references are
not set with that.

### CODE REVIEW

File: vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    public static Function<HasDescription<? extends VecAbstractLocalizedString>, Optional<String>> germanLanguageStringByDescription() {
+        return languageStringByDescription(VecLanguageCode.DE);
+    }
+
+    public static Function<HasDescription<? extends VecAbstractLocalizedString>, Optional<String>> englishLanguageStringByDescription() {
+        return languageStringByDescription(VecLanguageCode.EN);
+    }
+
+    public static Function<HasDescription<? extends VecAbstractLocalizedString>, Optional<String>> languageStringByDescription(
```
Comment (takebene) [inline vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java:57]: what do you think of the methode name "descriptionBy"

File: vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    public static Function<List<? extends VecAbstractLocalizedString>, Optional<String>> germanLanguageString() {
+        return languageString(VecLanguageCode.DE);
+    }
+
+    public static Function<List<? extends VecAbstractLocalizedString>, Optional<String>> englishLanguageString() {
+        return languageString(VecLanguageCode.EN);
+    }
+
+    public static Function<List<? extends VecAbstractLocalizedString>, Optional<String>> languageString(
```
Comment (takebene) [inline vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java:73]: by?

File: vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java (hunk 1 above)
Comment (DManstrator) [reply vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java:57]: I think `descriptionIn/With(VecLanguageCode languageCode)` makes more sense in this context. What do you think?

File: vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/ContentNavs.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    public static Function<VecContent, Optional<VecDocumentVersion>> documentVersionBy(
+            final String documentNumber) {
+        return content -> content.getDocumentVersions().stream()
+                .filter(documentVersion -> documentVersion.getDocumentNumber().equals(documentNumber))
+                .collect(StreamUtils.findOneOrNone());
+    }
+
+    // TODO Do this similar to HasRoles#getRolesWithType
+    public static <T extends VecUnit> Function<VecContent, Stream<T>> allUnitsOf(final Class<T> clazz) {
```
[the remark this answers is not in the document]
Comment (DManstrator) [reply vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/ContentNavs.java:64]: Done, thanks for the reminder.

File: vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java (hunk 2 above)
Comment (DManstrator) [reply vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java:73]: Same as above.

File: vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java (hunk 1 above)
Comment (takebene) [reply vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java:57]: yes you are right. i would say "in"

Comment (DManstrator) [reply vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java]: Another question would be if the typed methods could get a better name as well.

For example shortening `germanLanguageString` to `germanString` and `germanTypedStringBy`. What do you think of that?

**Review by takebene — APPROVED** [review]
ok

**Review by takebene — APPROVED** [review]
ok

**Pull request merged**